src/test/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfigurationTest.java

```diff
@@ -39,9 +39,17 @@
 @RunWith(MockitoJUnitRunner.class)
 public class BitbucketPluginConfigurationTest {
```

Comment:
I'm really keen to have this discussion, but I won't address this in this PR. I have a separate branch, `mh/cleanup-tests` that involves an audit and removal of `JenkinsRules` where providers can be used instead. I'm hoping we can discuss if it makes sense to separately package JenkinsRule tests there.

Reply:
+1 would love to rekindle this discussion and hear other team members' point of view. At this point, given a test that uses `JenkinsRule` is likely to take upwards of 10 seconds to run (not the only downside, but one factor IMO that can add up in the future), I think there's merit in revisiting this discussion at some point.

Reply:
> There is a benefit in keeping the tests that only need Jenkins out of the IT package as they are run during a normal "mvn verify" without the need to pass in a profile to get them to run.

I agree, _just_ moving any test that uses `JenkinsRule` to the IT package and _out of the way_ isn't probably the best course of action. I think it should be a combination of:
- Move tests that are genuinely ITs to the `it` package
- Re-write unit tests without the need for JenkinsRule, as much as possible without losing any unit test coverage. Unit tests are very valuable and removing them is never the right thing to do, so maybe we should reframe this discussion as "make unit tests really unit tests" rather than "move any unit test that uses JenkinsRule to ITs where they only run with the `it` profile".

The `JenkinsProvider` construct is an example of how we can bypass the need for JenkinsRule (and spinning up a Jenkins instance) in our unit tests.